crypto/evp/p_lib.c

```diff
@@ -356,6 +356,40 @@ int EVP_PKEY_set_type_str(EVP_PKEY *pkey, const char *str, int len)
 {
     return pkey_set_type(pkey, NULL, EVP_PKEY_NONE, str, len);
 }
+
+int EVP_PKEY_set_alias_type(EVP_PKEY *pkey, int type)
+{
+    const EVP_PKEY_ASN1_METHOD *ameth;
+
+    if (pkey == NULL) {
+        EVPerr(EVP_F_EVP_PKEY_SET_ALIAS_TYPE, EVP_R_NULL_ARGUMENT);
+        return 0;
+    }
+
+    if (pkey->type == type) {
+        return 1; /* it already is that type */
+    }
+
+    ameth = EVP_PKEY_asn1_find(NULL, type);
+
+    if (ameth == NULL) {
+        EVPerr(EVP_F_EVP_PKEY_SET_ALIAS_TYPE, EVP_R_UNSUPPORTED_ALGORITHM);
+        return 0;
+    }
```

Comment:
I think this ameth stuff can probably be removed now that you amended the "if" condition below to not need it.

Reply:
It stung my eye too. I mean it is redundant, because EVP_PKEY_type below calls EVP_PKEY_asn1_find and check for ameth being non-NULL.

And while you're at it, style guide suggests that block comments get stars aligned at opening star, i.e. and 4*n+1 position, not 4*n.